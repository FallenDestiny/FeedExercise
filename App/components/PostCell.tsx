import React, { Component } from 'react'
import { PostData } from '../redux/App/actions'
import { View, Image, Text, TouchableOpacity, StyleSheet } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { colors } from '../utils/colors'
import fonts from '../assets/fonts'
import Divider from './Divider'
import InfoView from './InfoView'
import images from '../assets/images'
import UserHeader from './UserHeader'

interface Props {
    post: PostData,
    onPress: () => void,
    onUserPress: () => void,
    testID?: string
}

export default function PostCell({ post, onPress, onUserPress, testID }: Props) {

    return (
        <TouchableOpacity testID={testID} activeOpacity={0.75} style={styles.cell} onPress={onPress}>
            {
                post && post.user_id ?
                    <UserHeader user={post.user_id[0]} date={post.createdAt} onPress={onUserPress} />
                    :
                    null
            }

            <View style={styles.body}>
                {
                    post && post.description ?
                        <Text ellipsizeMode='tail' numberOfLines={3} style={styles.textBody}>{post.description}</Text>
                        : null
                }

                {
                    post && post.image ?
                        <Image style={styles.mainImage} source={{ uri: post.image }} />
                        : null
                }

                {
                    post && post.likes && post.comments ?
                        <View style={styles.infoView}>
                            <InfoView icon={images.heart} text={post.likes.toString()} style={styles.info} />
                            <InfoView icon={images.message_circle} text={post.comments.toString()} style={styles.info} />
                        </View>
                        : null
                }
            </View>
            <Divider style={styles.divider} />
        </TouchableOpacity>
    )

}

const styles = StyleSheet.create({
    cell: {
        flex: 1,
        height: 360,
        marginVertical: 15,

    },
    body: {
        flex: 1,
        marginLeft: 53,

    },
    divider: {
        width: '100%',
        marginTop: 30
    },
    infoView: {
        height: 28,
        marginTop: 14,
        flexDirection: 'row'
    },
    mainImage: {
        flex: 2,
        borderRadius: 5,
        marginTop: 20,
        resizeMode: 'cover',
        width: undefined,
        height: undefined
    },
    textBody: {
        flex: 1,
        color: '#000000',
        fontFamily: fonts.sf_pro_display_light,
        fontSize: 16,
    },
    info: {
        height: '100%',
        marginRight: 15
    },
})
