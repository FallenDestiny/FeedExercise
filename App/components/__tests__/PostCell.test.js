import React from 'react';
import renderer from 'react-test-renderer';
import PostCell from '../PostCell'

const post = {
    "id": "1",
    "createdAt": "2020-02-18T04:40:49.601Z",
    "title": "Principal Identity Associate",
    "image": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "likes": 68,
    "comments": 43,
    "user_id": [
        {
            "id": "1",
            "postId": "1",
            "last_seen": "2020-02-18T06:33:57.162Z",
            "name": "Charles Hickle",
            "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bruno_mart/128.jpg",
            "followers": 10,
            "followings": 53,
            "description": "Extended logistical array",
            "posts": [
                {
                    "id": "1",
                    "createdAt": "2020-02-18T04:40:49.601Z",
                    "title": "Principal Identity Associate",
                    "image": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
                    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                    "likes": 68,
                    "comments": 43
                }
            ]
        }
    ]
}

describe('PostCell', () => {
    describe('Rendering', () => {
        it('draw all the cell data', () => {

            const component = renderer.create(<PostCell post={post} />).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw the cell with user only', () => {

            const component = renderer.create(<PostCell post={{user_id:post.user_id}} />).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw the cell with description only', () => {

            const component = renderer.create(<PostCell post={{description: post.description}} />).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw the cell with image only', () => {

            const component = renderer.create(<PostCell post={{image: post.image}} />).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw the cell with likes and comms only', () => {

            const component = renderer.create(<PostCell post={{likes: post.likes, comments:post.comments}} />).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw an empty cell', () => {
            const component = renderer.create(<PostCell/>).toJSON()
            expect(component).toMatchSnapshot()
        });
    });

})
