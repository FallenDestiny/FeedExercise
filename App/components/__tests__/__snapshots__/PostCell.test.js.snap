// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCell Rendering draw all the cell data 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "height": 360,
      "marginVertical": 15,
      "opacity": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "maxHeight": 55,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/bruno_mart/128.jpg",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "borderRadius": 27.5,
          "height": "100%",
          "resizeMode": "contain",
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#151522",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 13,
            "marginLeft": 15,
          }
        }
      >
        Charles Hickle
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#151522",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 13,
              "marginLeft": 15,
            },
            Object {
              "color": "#999999",
            },
          ]
        }
      >
        9 days ago
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 53,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontFamily": "SFProDisplay-Light",
          "fontSize": 16,
        }
      }
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </Text>
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "flex": 2,
          "height": undefined,
          "marginTop": 20,
          "resizeMode": "cover",
          "width": undefined,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 28,
          "marginTop": 14,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "height": "100%",
              "marginRight": 15,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/assets/images/heart.png",
            }
          }
          style={
            Object {
              "aspectRatio": 1,
              "height": "100%",
              "resizeMode": "contain",
              "width": undefined,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 15,
                "marginLeft": 5,
              },
              undefined,
            ]
          }
        >
          68
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "height": "100%",
              "marginRight": 15,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/assets/images/message-circle.png",
            }
          }
          style={
            Object {
              "aspectRatio": 1,
              "height": "100%",
              "resizeMode": "contain",
              "width": undefined,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 15,
                "marginLeft": 5,
              },
              undefined,
            ]
          }
        >
          43
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        Object {
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`PostCell Rendering draw an empty cell 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "height": 360,
      "marginVertical": 15,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 53,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        Object {
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`PostCell Rendering draw the cell with description only 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "height": 360,
      "marginVertical": 15,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 53,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontFamily": "SFProDisplay-Light",
          "fontSize": 16,
        }
      }
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        Object {
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`PostCell Rendering draw the cell with image only 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "height": 360,
      "marginVertical": 15,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 53,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "flex": 2,
          "height": undefined,
          "marginTop": 20,
          "resizeMode": "cover",
          "width": undefined,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        Object {
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`PostCell Rendering draw the cell with likes and comms only 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "height": 360,
      "marginVertical": 15,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 53,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 28,
          "marginTop": 14,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "height": "100%",
              "marginRight": 15,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/assets/images/heart.png",
            }
          }
          style={
            Object {
              "aspectRatio": 1,
              "height": "100%",
              "resizeMode": "contain",
              "width": undefined,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 15,
                "marginLeft": 5,
              },
              undefined,
            ]
          }
        >
          68
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "height": "100%",
              "marginRight": 15,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/assets/images/message-circle.png",
            }
          }
          style={
            Object {
              "aspectRatio": 1,
              "height": "100%",
              "resizeMode": "contain",
              "width": undefined,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#999999",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 15,
                "marginLeft": 5,
              },
              undefined,
            ]
          }
        >
          43
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        Object {
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`PostCell Rendering draw the cell with user only 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "height": 360,
      "marginVertical": 15,
      "opacity": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "maxHeight": 55,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/bruno_mart/128.jpg",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "borderRadius": 27.5,
          "height": "100%",
          "resizeMode": "contain",
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#151522",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 13,
            "marginLeft": 15,
          }
        }
      >
        Charles Hickle
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 53,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        Object {
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
